subsys/net/ip/igmp.c

```diff
@@ -114,13 +243,15 @@ static int igmp_send(struct net_pkt *pkt)
 	return 0;
 }
 
+#if !defined(CONFIG_NET_IPV4_IGMPV3)
```

Comment:
Could / should we support both v2 and v3 at the same time?

Reply:
Hmm indeed, we should be able to respond to queries from older versions. I will take a look!

Reply:
I added the detection of the IGMP version of a received query. Now, IGMPv2 queries are also answered when IGMPv3 is enabled.